### PR TITLE
Refactor JMAP Client

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: james/auth.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	identityClient *v1alpha1.IdentityV1alpha1Client
+	once           sync.Once
+)
+
+func getJamesToken() (string, error) {
+	{ //TODO: remove this block of code
```
Comment (Imtiaz246) [inline james/auth.go]: Remove these comments.

File: james/jmap.go [hunk 2]
```diff
@@ -229,3 +234,179 @@ func (j *JmapClient) initializeMailboxIds() error {
...
+
+func (j *JMAPClient) uploadAttachment(attachmentName string, blob io.Reader) (*email.BodyPart, error) {
+	resp, err := j.Upload(j.userId, blob)
+	if err != nil {
+		return nil, err
+	}
+
+	myAttachment := email.BodyPart{
+		BlobID: resp.ID,
+
```
Comment (Imtiaz246) [inline james/jmap.go]: Remove need these newlines.

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: james/auth.go (hunk 1 above)
Comment (samiulsami) [reply james/auth.go]: Fixed.

File: james/jmap.go (hunk 2 above)
Comment (samiulsami) [reply james/jmap.go]: Fixed.

File: james/jmap.go [hunk 3]
```diff
@@ -229,3 +234,163 @@ func (j *JmapClient) initializeMailboxIds() error {
...
 }
+
+func (j *JMAPClient) GetTestRecipient() (string, error) {
+	j.muJmap.RLock()
+	defer j.muJmap.RUnlock()
+	return Recipient, nil
+}
+
+/*
+-----------------------------------------
```
Comment (Imtiaz246) [inline james/jmap.go]: Remove these comments

File: james/auth.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+var (
+	identityClient *v1alpha1.IdentityV1alpha1Client
+	once           sync.Once
+)
+
+func getJamesToken() (string, error) {
+	{
+		if token, ok := os.LookupEnv("INBOX_SERVER_TOKEN"); ok && len(token) > 0 {
```
Comment (Imtiaz246) [inline james/auth.go:24]: Remove comment

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

**Pull request merged**